From bda12f64e60f54438a6af64c328c69b426e3cf8c Mon Sep 17 00:00:00 2001
From: Ofir Fedida <ofedida@marvell.com>
Date: Tue, 12 Apr 2016 14:14:58 +0300
Subject: [PATCH 168/239] mv_ddr: atf: Add reg_bit_clrset func

This patch replaces mmio_clrsetbits2_32 function to
reg_bit_clrset.

Change-Id: I74c66b3479b729b4d42b1e25528d093d8bdd543d
Signed-off-by: Ofir Fedida <ofedida@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28965
Reviewed-by: Victor Axelrod <victora@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Nadav Haklai <nadavh@marvell.com>
---
 drivers/marvell/mv_ddr/mv_ddr_atf_wrapper.h | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/marvell/mv_ddr/mv_ddr_atf_wrapper.h b/drivers/marvell/mv_ddr/mv_ddr_atf_wrapper.h
index fe3ef50..df4109f 100644
--- a/drivers/marvell/mv_ddr/mv_ddr_atf_wrapper.h
+++ b/drivers/marvell/mv_ddr/mv_ddr_atf_wrapper.h
@@ -259,19 +259,18 @@ static inline void reg_bit_clr(u32 addr, u32 mask)
 		      mmio_read_32(INTER_REGS_BASE + addr) & ~mask);
 }
 
-static inline void mmio_write2_32(u32 val, u32 addr)
+static inline void reg_bit_clrset(u32 addr, u32 val, u32 mask)
 {
-	mmio_write_32(addr, val);
+	mmio_clrsetbits_32(INTER_REGS_BASE + addr, mask, val);
 }
 
-static inline void mmio_clrsetbits2_32(u32 addr, u32 val, u32 mask)
+static inline void mmio_write2_32(u32 val, u32 addr)
 {
-	mmio_clrsetbits_32(addr, mask, val);
+	mmio_write_32(addr, val);
 }
 
 #define writel mmio_write2_32
 #define readl mmio_read_32
-#define writel_clrset mmio_clrsetbits2_32
 
 #if defined(CONFIG_DDR4)
 #define vref_calibration_wa vrefCalibrationWA
-- 
1.9.1

